### PR TITLE
Use meson to build the project

### PR DESCRIPTION
**Pull request by mwperina**
Use meson to build the project and run meson build in GH CI.

Signed-off-by: Martin Perina <mperina@redhat.com>

### CODE REVIEW

**Comment by mwperina** [comment]
Still only a draft:

* I haven't figured out yet the things about formatting
* No idea if multiple meson.build files are good or if we should use single meson.build file for the whole project

**Comment by engelmi** [comment]
> Still only a draft:
> 
> * I haven't figured out yet the things about formatting
> * No idea if multiple meson.build files are good or if we should use single meson.build file for the whole project

Nice work! 
`systemd` uses multiple meson.build files as well. Its also much more readable and easier to understand having multiple smaller files instead of a huge one. So I think the multiple meson files are very good. 
For the formatting, I'd like to keep the `Makefile` so that we can use a simple `make fmt` and `make fmt-check` instead of the longer command. The content we can replace with the ninja cmd - which is better than the script-ish lines I hacked down before. What do you think?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -27,11 +27,18 @@ jobs:
       image: quay.io/centos/centos:stream9
 
     steps:
+      - name: Enable additional repositories
+        run: |
+          dnf install -y dnf-plugin-config-manager
+
+          # CRB is required for meson
+          dnf config-manager -y --set-enabled crb
```
Comment (engelmi) [inline .github/workflows/build.yml:35]: This did not work for me. I had to do
```bash
sudo subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-rpms
```
as described in https://access.redhat.com/articles/4348511

File: README.md [hunk 2]
```diff
@@ -12,44 +12,82 @@ TBD
...
 ### Code Style
 
-`clang-format` is used to enforce a unified code style. For setup instructions, please refer to the [official Clang getting started guide](https://clang.llvm.org/get_started.html). 
+`clang-format` is used to enforce a unified code style. For setup instructions, please refer to the [official Clang getting started guide](https://clang.llvm.org/get_started.html). `clang-format` is integrated into the
+[meson](https://mesonbuild.com/) build system.
 
 Once `clang-format` is set up, all source files can be formatted via:
 ```bash
-make fmt
+meson setup builddir
```
Comment (engelmi) [inline README.md]: If we stick with the name `builddir`, it should be added to `.gitignore` and the existing `bin` can be removed.

File: .github/workflows/build.yml (hunk 1 above)
Comment (mwperina) [reply .github/workflows/build.yml:35]: subscription-manager is only for RHEL, it shouldn't be used upstream

File: README.md (hunk 2 above)
Comment (mwperina) [reply README.md]: Good idea!

Comment (mwperina) [reply README.md]: Done

**Comment by mwperina** [comment]
> > Still only a draft:
> > 
> > * I haven't figured out yet the things about formatting
> > * No idea if multiple meson.build files are good or if we should use single meson.build file for the whole project
> 
> Nice work! `systemd` uses multiple meson.build files as well. Its also much more readable and easier to understand having multiple smaller files instead of a huge one. So I think the multiple meson files are very good. For the formatting, I'd like to keep the `Makefile` so that we can use a simple `make fmt` and `make fmt-check` instead of the longer command. The content we can replace with the ninja cmd - which is better than the script-ish lines I hacked down before. What do you think?

Using ninja with clang-format doesn't produce that nice results as using make, so returning back to use make around formatting

**Review by sdunnagan — APPROVED** [review]

**Pull request merged**